lib/libzrepl/data_conn.c

```diff
@@ -725,9 +725,9 @@ uzfs_zvol_io_conn_acceptor(void *arg)
 #ifdef DEBUG
 			hbuf = kmem_alloc(sizeof (NI_MAXHOST), KM_SLEEP);
 			sbuf = kmem_alloc(sizeof (NI_MAXSERV), KM_SLEEP);
-			rc = getnameinfo(&in_addr, in_len, hbuf, sizeof (hbuf),
-			    sbuf, sizeof (sbuf), NI_NUMERICHOST |
-			    NI_NUMERICSERV);
+			rc = getnameinfo(&in_addr, in_len, hbuf,
+			    sizeof (NI_MAXHOST), sbuf, sizeof (NI_MAXSERV),
+			    NI_NUMERICHOST | NI_NUMERICSERV);
```

Comment:
You need to remove sizeof in front of NI_MAXHOST and NI_MAXSERV and the same for kmem_allocs above. Those are integers and not C types.